obspy/signal/cross_correlation.py

```diff
@@ -76,7 +76,44 @@ def _xcorr_slice(a, b, shift, domain='freq'):
     return c[mid - shift:mid + shift + len(c) % 2]
 
 
-def correlate(a, b, shift, demean=True, normalize=True, domain='freq'):
+def _normxcorr(a, b, shift, domain='freq'):
+    """
+    Normalized cross-correlation
+    """
+    mid = (len(a) + len(b) - 1) // 2
+    if shift is None:
+        shift = mid
+    # Normalize short window before correlation
+    b = (b - b.mean()) / (b.std() * len(b))
+    # We HAVE to pad in this case
+    a = np.pad(a, pad_width=len(b), mode='constant', constant_values=0)
+    if domain == 'time':
+        c = scipy.signal.correlate(a, b, 'valid')
+    else:
+        c = scipy.signal.fftconvolve(a, b[::-1], 'valid')
+    # Calculate moving mean and moving standard deviation
+    # using cumulative sums.
+    c = c[1:-1]
+    move_mean = np.cumsum(a)
+    move_mean = move_mean[len(b):-1] - move_mean[:-len(b) - 1]
+    numerator = c - move_mean * b.mean()
+    denominator = np.cumsum(a ** 2)
+    denominator = denominator[len(b):-1] - denominator[:-len(b) - 1]
+    np.multiply(move_mean, move_mean, out=move_mean)
+    np.divide(move_mean, len(a), out=move_mean)
```

Comment:
Found it! This should be len(b).

Reply:
Indeed, for some reason I had this condition in an earlier version, but I reworked it and removed it - added the switch back in! (to be pushed!)